### PR TITLE
TM-5587: Manage EL Position Filtering

### PR DESCRIPTION
**Pull request by pveland621**
ACs:
1. Verify query being sent back to API is correct (should be a dict of user-inputted filters + values)

Dual Merge: 
- [API PR](https://github.com/MetaPhase-Consulting/State-TalentMAP-API/pull/949)

[Ticket](https://metaphase.atlassian.net/browse/TM-5587)

### CODE REVIEW

File: src/Components/AdministratorPage/ManageEntryLevel/ManageEntryLevel.jsx [hunk 1]
```diff
@@ -70,22 +72,21 @@ const ManageEntryLevel = () => {
...
+  //   selectedTps,
+  //   selectedBureaus,
+  //   selectedOrgs,
+  //   selectedGrade: selectedGrades,
+  //   selectedSkills,
+  //   selectedJobs,
+  //   selectedLanguage: selectedLanguages,
+  //   overseas,
+  //   domestic,
+  // });
```
Comment (M00staff) [inline src/Components/AdministratorPage/ManageEntryLevel/ManageEntryLevel.jsx:86]: so, i take it that getting the filters working does not cover saving the user selections in redux?

is that by design?  

its a subtask on the ticket, so i would assume so

File: src/Components/AdministratorPage/ManageEntryLevel/ManageEntryLevel.jsx [hunk 2]
```diff
@@ -53,7 +55,7 @@ const ManageEntryLevel = () => {
     'el-orgs': selectedOrgs.map(orgObject => (orgObject?.code)),
     'el-grades': selectedGrades.map(gradeObject => (gradeObject?.code)),
     'el-skills': selectedSkills.map(skillObject => (skillObject?.code)),
-    'el-jobs': selectedJobs.map(jobObject => (jobObject?.id)),
+    'el-jobs': selectedJobs.map(jobObject => (jobObject?.code)),
```
Comment (M00staff) [inline src/Components/AdministratorPage/ManageEntryLevel/ManageEntryLevel.jsx:58]: so, the domestic & overseas flags aren't a part of the request or call for the data?

then what do they do?

File: src/Components/AdministratorPage/ManageEntryLevel/ManageEntryLevel.jsx [hunk 3]
```diff
@@ -100,13 +101,15 @@ const ManageEntryLevel = () => {
       overseas,
       domestic,
     ];
-    if (filters.flat().length === 0) {
+    // Don't reload positions if there are no user selected filters AND
+    // OD/Domestic is false
+    if (filters.flat().length === 2 && !filters.flat().includes(true)) {
```
Comment (M00staff) [inline src/Components/AdministratorPage/ManageEntryLevel/ManageEntryLevel.jsx:106]: this logic is incorrect - when you click the checkbox for `overseas` or `domestic` the value for it changes from a boolean to a string - then doesn't ever change back to anything else after being unchecked

image
image
image

Prolly these `onChanges` need to be something like `(e) => setOverseas(e)` or something or other ... also sometimes i have to use `onCheckBoxClick` 
image

**Review by M00staff — CHANGES_REQUESTED** [review]
the logic for the overseas & domestic checkboxes are off - i dont think this bit of feature functions as intended

File: src/Components/AdministratorPage/ManageEntryLevel/ManageEntryLevel.jsx (hunk 2 above)
Comment (M00staff) [reply src/Components/AdministratorPage/ManageEntryLevel/ManageEntryLevel.jsx:58]: i also see that the domestic flag has a strike through in the spreadsheet ... were we supposed to remove it?

image

File: src/Components/AdministratorPage/ManageEntryLevel/ManageEntryLevel.jsx (hunk 3 above)
Comment (M00staff) [reply src/Components/AdministratorPage/ManageEntryLevel/ManageEntryLevel.jsx:106]: only hitting clear filters resets it to false 🤔

File: src/Components/AdministratorPage/ManageEntryLevel/ManageEntryLevel.jsx (hunk 1 above)
Comment (pveland621) [reply src/Components/AdministratorPage/ManageEntryLevel/ManageEntryLevel.jsx:86]: this ticket only captures filtering, updated ticket

File: src/Components/AdministratorPage/ManageEntryLevel/ManageEntryLevel.jsx (hunk 3 above)
Comment (pveland621) [reply src/Components/AdministratorPage/ManageEntryLevel/ManageEntryLevel.jsx]: reworked checkboxes to be bools only and added FE/BE support for adding domestic/overseas to the query

File: src/Components/AdministratorPage/ManageEntryLevel/ManageEntryLevel.jsx (hunk 2 above)
Comment (pveland621) [reply src/Components/AdministratorPage/ManageEntryLevel/ManageEntryLevel.jsx:58]: basically WS returns overseas or domestic as one field `POS_OVERSEAS_DESC` which will be either `Overseas` or `Domestic` so I think Bob just went and crossed that out for clarity

File: src/Components/AdministratorPage/ManageEntryLevel/ManageEntryLevel.jsx (hunk 1 above)
Comment (elizabeth-jimenez) [reply src/Components/AdministratorPage/ManageEntryLevel/ManageEntryLevel.jsx:88]: heads up - user selections are caching of filters user has selected

Comment (pveland621) [reply src/Components/AdministratorPage/ManageEntryLevel/ManageEntryLevel.jsx]: gotcha, was misunderstanding the function. put saving selections to redux back in

**Review by M00staff — APPROVED** [review]
double checked the request objects in swagger and clicked through ... looks like its working - you're all set!

**Pull request merged**